### PR TITLE
[fmiweather] Fix compiler warnings and SAT issues

### PR DESCRIPTION
**Pull request by jlaur**
This leaves only two compiler warnings that I was not able to resolve - see comments.

### CODE REVIEW

File: bundles/org.openhab.binding.fmiweather/src/main/java/org/openhab/binding/fmiweather/internal/AbstractWeatherHandler.java [hunk 1]
```diff
@@ -97,14 +96,19 @@ public void handleCommand(ChannelUID channelUID, Command command) {
...
-                if (prevFuture == newFuture) {
+                if (isSameFuture(prevFuture, newFuture)) {
                     logger.trace("REFRESH received. Previous refresh ongoing, will wait for it to complete in {} ms",
                             lastRefreshMillis + REFRESH_THROTTLE_MILLIS - System.currentTimeMillis());
                 }
             }
         }
     }
 
+    @SuppressWarnings("PMD.CompareObjectsWithEquals")
```
Comment (jlaur) [inline bundles/org.openhab.binding.fmiweather/src/main/java/org/openhab/binding/fmiweather/internal/AbstractWeatherHandler.java:107]: Now isolated here, but still warning:
```
[WARNING] AbstractWeatherHandler.java:[107,23] Unsupported @SuppressWarnings("PMD.CompareObjectsWithEquals")
```

Only needed for trace logging, but didn't want to change this. Also did not want to risk refactoring `AtomicReference` logic for this reason.

File: bundles/org.openhab.binding.fmiweather/src/main/java/org/openhab/binding/fmiweather/internal/client/Client.java [hunk 2]
```diff
@@ -91,12 +88,12 @@ public class Client {
 
         @SuppressWarnings("rawtypes")
         @Override
-        public @Nullable Iterator getPrefixes(@Nullable String val) {
+        public @Nullable Iterator getPrefixes(@Nullable String namespaceURI) {
```
Comment (jlaur) [inline bundles/org.openhab.binding.fmiweather/src/main/java/org/openhab/binding/fmiweather/internal/client/Client.java:91]: I had a hard time here, but the warning remains:
```
[WARNING] Client.java:[91,26] Type safety: The return type java.util.Iterator for getPrefixes(java.lang.String) from the type new javax.xml.namespace.NamespaceContext(){} needs unchecked conversion to conform to java.util.Iterator<java.lang.String> from the type javax.xml.namespace.NamespaceContext
```

First step was to change to correct method signature:
```java
public @Nullable Iterator<String> getPrefixes(@Nullable String namespaceURI) {
```

But now having null annotation issues. I tried all combinations of `@Nullable`/`@NonNull`, but none of them were accepted.

**Review by lsiepel — APPROVED** [review]
Thanks, LGTM

File: bundles/org.openhab.binding.fmiweather/src/main/java/org/openhab/binding/fmiweather/internal/client/Client.java (hunk 2 above)
Comment (lsiepel) [reply bundles/org.openhab.binding.fmiweather/src/main/java/org/openhab/binding/fmiweather/internal/client/Client.java:91]: I have faced similar challanges and finally gave up. I think it is acceptable to have a few warnings like this. We have like 10k other warnings to fix ;-) Escpecially when null annotations are added to all bindings. Must say we do make progress on this subject, slowly, but progress!

**Pull request merged**